mlir-tensorrt/executor/lib/Runtime/API/API.cpp

```diff
@@ -463,6 +466,22 @@ void AllocTracker::track(PointerInfo info) {
     map.insert(std::make_pair(info.ptr, std::move(value)));
     return;
   }
+
+  auto existingReferenceCount = map.at(info.ptr)->externalReferenceCount.load();
+
+  // If exisiting value in map is internally maanged and not yet released
+  // internally, we can just increment the reference count for the exisiting
+  // entry.
+  if (get(info.ptr).isInternallyManaged() &&
+      !map.at(info.ptr)->releasedInternally && existingReferenceCount > 0 &&
+      info.isExternallyManaged()) {
+    ++map.at(info.ptr)->externalReferenceCount;
+    return;
+  }
+
+  // In the default case when there is an exisiting value in map, we reuse the
+  // exisiting reference count for the new pointer.
+  value->externalReferenceCount = existingReferenceCount;
```

Comment:
are we expecting the incoming pointer reference count to be 0 here? Should we assert that invariant?